python/cudf/cudf/tests/test_list.py

```diff
@@ -246,3 +246,51 @@ def test_get_nulls():
     with pytest.raises(IndexError, match="list index out of range"):
         sr = cudf.Series([[], [], []])
         sr.list.get(100)
+
+
+@pytest.mark.parametrize(
+    "data, value, expect",
+    [
+        ([[1, 2, 3], [3, 4, 5], [4, 5, 6]], 4, [False, True, True]),
+        (
+            [[1, 2.0, 3.0], [3.0, 4.0, 5], [4.0, 5, 6.0]],
+            4.0,
+            [False, True, True],
+        ),
+        (
+            [["a", "b", "c"], ["b", "e", "f"], ["e", "g", "m"]],
+            "b",
+            [True, True, False],
+        ),
+    ],
+)
+def test_contains_scalar(data, value, expect):
+    sr = cudf.Series(data)
+    search_key = cudf.Scalar(value, sr.dtype.element_type)
+    expect = cudf.Series(expect)
+    got = sr.list.contains(search_key)
+    assert_eq(expect, got)
+
+
+def test_contains_invalid_search_key():
+    sr = cudf.Series([[1.0, 2.0, 3.0], [], [5.0, 7.0, 8.0]])
+    search_key = cudf.Scalar(4.0, sr.dtype.element_type)
+    expect = cudf.Series([False, False, False])
+    got = sr.list.contains(search_key)
+    assert_eq(expect, got)
+
+
+def test_contains_null_search_key():
+    sr = cudf.Series([[1, 2, 3], [3, 4, 5], [4, 5, 6]])
+    search_key = cudf.Scalar(cudf.NA, sr.dtype.element_type)
+    expect = cudf.Series([False, False, False])
+    got = sr.list.contains(search_key)
+    assert_eq(expect, got)
```

Comment:
Connected offline and cannot think of a better way to mimic this behavior as of now. I'm happy if python currently just returns what libcudf returns, which is `[<NA>, <NA>, <NA>]` until someone raises an issue about it.